time_needed_to_rearrange_a_binary_string.cpp

```diff
@@ -0,0 +1,57 @@
+// Author: Hacer Akıncı
+// Reviewer: Ömer Faruk Erdem
+// Reviewer: Fatih Baskın
+// Question Link: https://leetcode.com/problems/time-needed-to-rearrange-a-binary-string/
+
+/*
+    We count "zeros" to '1'. If we have a 0001 string. 3 zeros means 3 seconds needed to make it 1000.
+    Also, we need at least one more second for every '1', but not less than "zero" seconds. 
+    For example, if we have 000111, we need 5 seconds(two more seconds for two more "ones") to turn 111000.  
+    
+    Here is calculation for "0110100001" test case:
+
+            | 0 | 1 | 1 | 0 | 1 | 0 | 0 | 0 | 0 | 1 |
+    zeros     1   1   1   2   2   3   4   5   6   6
+    seconds   0   1   2   2   3   3   3   3   3   6
+                              ^                   
+                              |                    
+    For i == 4  we need 3 seconds, even though there are two zero before 4th index.
+    Because we need an extra second for '1' in 2nd index as I mentioned and this affects the rest of the string. 
+    
+            | 0 | 1 | 1 | 0 | 1 | 0 | 0 | 0 | 0 | 1 |
+    zeros     1   1   1   2   2   3   4   5   6   6
+    seconds   0   1   2   2   3   3   3   3   3   6
+                                                  ^
+                                                  |
+    For i == 9  we need 6 seconds, even though we need 3 seconds for i==8.

```

Comment:
I have objections regarding using == in plain language. "For i == 9" reads as "for if i is equal to 9", it does not make any sense. Second, even if you choose to use symbols, decide on the format. Either write `i=9` and `i=8` or `i = 9` and `i = 8`.